### PR TITLE
Readthedocs

### PR DESCRIPTION
**Pull request by KPHippe**
Here is a preliminary readthedocs implementation

Includes: 
* autosummary of ot2util package. Anytime we add a *documented* module/class etc to the package, a corresponding entry in the documentation will be made, just need to rebuild documents (`make html` command) 
* basic YAML guide. I am not sure if this is written well, but it shows a couple of examples and might be helpful. please suggest any improvements/additions and I will be sure to do that

Things still to do: 
* Documentation on writing your own algorithm to run on OT2, wanted to wait for implementation of algorithm to be decided on. Or I can write what went into the grid_search.py, whatever works best

### CODE REVIEW

**Comment by KPHippe** [comment]
Ended up rewriting a local/remote config for the experiments to show both. This is the reason for the merge conflict

**Comment by braceal** [comment]
TODO: 
- [ ] Put docs dependencies in dev requirements
- [ ] Put yaml field docstrings in pydantic classes.

**Comment by KPHippe** [comment]
Should be ready for merge. `autodoc_pydantic` cleaned up the pydantic documentation really nicely. Updated the field documentation. Should be ready to go

File: docs/source/pages/usage/yaml.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+Writing an Algorithm YAML
+################################
+
+Running a Simulation Algorithm
+******************************
+
+Fields 
+^^^^^^^
+    - :code:`remote_dir: Optional[Path]` a pathlike object pointing to place to stage experiments on remote OT2 pointing
```
Comment (braceal) [inline docs/source/pages/usage/yaml.rst:12]: We can clean these up to avoid duplication. Link to pydantic config models

**Comment by KPHippe** [comment]
Just tweaked the YAML file to make it reference some of the documentation a bit less brittle. Also started a (rough) Algorithm example and explanation.

**Pull request merged**